app/stac/services/stac_item_creator.py

```diff
@@ -95,10 +98,11 @@ def _generate_and_add_metadata(self, filepath, add_asset=True):
         self.generated_rio_stac_items.append(generated_stac)
 
         if add_asset:
+            asset_key = parser.get_asset_common_name_from_filename(return_asset_name(filepath))
```

Comment:
Why are we putting this outside of the os parser? This will fail for the example parser and is now not generic.